tests/stdlib/algorithm/trepr.nim

```diff
@@ -67,6 +67,20 @@ template main() =
     else:
       doAssert reprOpenarray(arr) == "[1, 2, 3]"
 
+  block repr_with_range:
+    type
+      Range = range[1..2]
+      DistinctRange = distinct range[1..2]
+      Nested = distinct range[DistinctRange(1)..DistinctRange(2)]
+
+    # simple case: only a range
+    doAssert repr(Range(1)) == "1"
+    # more complex: a range wrapped in a ``distinct``
+    doAssert repr(DistinctRange(1)) == "1"
+    # complex case: a distinct range where the base type is also a distinct
+    # type
+    doAssert repr(Nested(DistinctRange(1))) == "1"
+
```

Comment:
This will be caught via runnable examples in CI, although much later..

Reply:
Could you point me to the `runnableExamples`? I can't seem to find the one that tests this case.

Reply:
Sorry, I should have confirmed it, but I got the false impression that the rangebase runnable examples were equivalent. Brain was doing fast and loose pattern matching based on the setup (wasn't conscious of it).

You're right, these are different story about that.